hs-abci-examples/nameservice/src/Nameservice/Modules/Nameservice/Messages.hs

```diff
@@ -17,14 +22,53 @@ data MsgSetName =  MsgSetName
   , msgSetNameOwner :: Address
   }
 
+-- TL;DR. ValidateBasic: https://cosmos.network/docs/tutorial/set-name.html#msg
+fromProtoMsgSetName :: M.SetName -> Either Text MsgSetName
```

Comment:
the `MsgSetName` prefix is internal, similarly with `MsgDeleteName` etc. Don't include them in the error messages